src/components/input/input.js

```diff
@@ -70,22 +78,25 @@ export const Input = ({
       <div className={controlClassNames.join(" ")}>
         <input
           className={inputClasses}
-          type={type}
+          type={typeState}
           name={name}
           disabled={disabled}
           defaultValue={defaultValue}
           placeholder={usePlaceholder ? label : ""}
           onChange={onChange}
           required={required}
-          min={type === "number" && min ? min : void 0}
-          max={type === "number" && max ? max : void 0}
-          step={type === "number" && step ? step : void 0}
+          min={typeState === "number" && min ? min : void 0}
+          max={typeState === "number" && max ? max : void 0}
+          step={typeState === "number" && step ? step : void 0}
```

Comment:
I believe these could be left as `type`.

Reply:
It can't be type because it's prop from parent. It won't be updated, that's why I used state for live changing inside Input component.

Reply:
The only time the typeState changes is for password / text combo, not number.

I'm fine with keeping this as typeState, but then let's change it on change bellow as well.